### PR TITLE
Fix registration

### PR DESCRIPTION
**Pull request by valerauko**
Registration didn't seem to work from the API (mostly due to schema changes not being reflected to the spec).

## Changes
* fix `screenname` typos (involves a migration!)
* adjust spec to reflect new columns
* add `active` and `token` to the `create-user!` sql
* authenticate user after registration

## Questions
* Is there any better way to get the user after registration?

### CODE REVIEW

**Comment by valerauko** [comment]
Oh. Okay. I'll do that then.

**Comment by yogthos** [comment]
And registration may have email step [here](https://github.com/yogthos/krueger/blob/39584859830d76df2181020a2edb3cbcf72e83fd/src/clj/krueger/routes/services/auth.clj#L64), so trying to authenticate after creating the user is probably not the way to go in that case.

**Comment by yogthos** [comment]
Probably just returning something like `(ok {:result :confirmation-pending})` or something would be the way to go. Then the client notifies the user that they need to finish registration using the link in the confirmation email.

**Comment by valerauko** [comment]
Made it return `common/Success` for now.

File: src/clj/krueger/routes/services.clj [hunk 1]
```diff
@@ -48,17 +48,16 @@
...
                   pass :- s/Str]
     :summary "user registration handler"
-    (auth/register!
-      {:screnname screnname
-       :email     email
-       :pass      pass}
-      req)
-    (ok {:result :ok}))
+    ; REVIEW: I'm not sure this gives the correct result if `register!` 500s
+    (ok {:result (auth/register! {:screenname screenname
```
Comment (yogthos) [inline src/clj/krueger/routes/services.clj]: The default middleware [here](https://github.com/yogthos/krueger/blob/a8e3814c9be7ba7d3495a6b069ccf46ab8e54f06/src/clj/krueger/middleware.clj#L39) should handle that case gracefully.

**Pull request merged**